MCH: moved time calibration to digits filter

aferrero2707 opened the pull request:
The time calibration is controlled by the `MCHDigitFilterParam.timeOffset` configuration parameter, that specifies the global offset (in BC units) applied to the time stamp of digits and ROFs.

aphecetche commented:
@aferrero2707, if you agree, I would propose that : 

1 - we merge https://github.com/AliceO2Group/AliceO2/pull/8737 to get the `DigitFilterParam` in place
2 - we introduce the digit noise filtering (PR to be opened after 1 is merged, see https://github.com/AliceO2Group/AliceO2/commit/65b0fb905883fbc501cab7e2d763996030145149) 
3 - we rebase this one on (1 or 2) to benefit from the DigitFilterParam

or equivalently, 1 - 3 -2 (2 and 3 being different filtering features)

what do you think ?

aphecetche commented:
@aferrero2707, if you agree, I would propose that : 

1 - we merge https://github.com/AliceO2Group/AliceO2/pull/8737 to get the `DigitFilterParam` in place
2 - we introduce the digit noise filtering (PR to be opened after 1 is merged, see https://github.com/AliceO2Group/AliceO2/commit/65b0fb905883fbc501cab7e2d763996030145149) 
3 - we rebase this one on (1 or 2) to benefit from the DigitFilterParam

or equivalently, 1 - 3 -2 (2 and 3 being different filtering features)

what do you think ?

aferrero2707 commented:
I think 1 and 2 are more urgent than this PR, so we can rebase this one once the other two are merged.

shahor02 commented:
> I think 1 and 2 are more urgent than this PR, so we can rebase this one once the other two are merged.

@aferrero2707 OK, would you check/validate https://github.com/AliceO2Group/AliceO2/pull/8737 ? I'll merge it once fullCI is passed

aferrero2707 committed:
[MCH] moved time calibration to digits filter

The commit introduces a step in the digits filtering that
allows to shift the time of the digits and ROFs by a
common offset specified in the

    DigitFilterParam.timeOffset

configuration parameter.

aferrero2707 commented:
@aphecetche all checks are green, could you please have a look if the modifications are OK for you? meanwhile I will commit a small update of the documentation.
We should probably also consider a unit test dedicated to the digits filter, but I would propose to make a dedicated PR for that.

aferrero2707 committed:
[MCH] updated digit filtering documentation

pillot commented on Detectors/MUON/MCH/DigitFiltering/src/DigitFilteringSpec.cxx:
```diff
@@ -125,6 +144,8 @@ class DigitFilteringTask
   bool mSanityCheck;
   bool mUseMC;
   DigitFilter mIsGoodDigit;
+  std::string mTimeCalibSource;
```
this data member is not used

aferrero2707 replied:
Removed.

pillot reviewed (commented):
Hi @aferrero2707 ,
Appart from the little change below, shouldn't the configurable parameter CoDecParam::Instance().sampaBcOffset be removed?
In which case it also has to be removed from the few places where it is still used (e.g. testSampaBunchCrossingCounter.cxx, codec-closure-test.sh and testClosureCoDecDigit.cxx).

aferrero2707 committed:
[MCH] removed unused parameters

aferrero2707 commented:
@pillot you are absolutely right! It seems I stopped in the middle of the code cleanup to remove the `sampaBcOffset`.
